app/models/preupgrade_report.rb

```diff
@@ -3,21 +3,16 @@
 class PreupgradeReport < ::Report
   has_many :preupgrade_report_entries, dependent: :destroy
 
-  def self.create_report(host, date, status, data)
-    report = PreupgradeReport.create(:host => host,
-                                     :reported_at => date,
-                                     :status => status)
-    data[:entries]&.each do |entry|
-      data = entry.slice(:title, :actor, :summary, :audience, :severity)
-                  .merge(:preupgrade_report => report,
-                         :hostname => host.hostname,
-                         :tags => entry[:tags].to_json,
-                         :detail => entry[:detail].to_json,
-                         :leapp_run_id => data[:leapp_run_id])
-      data.permit!
-      PreupgradeReportEntry.create! data
+  def self.create_report(host, data)
+    report = PreupgradeReport.create(host: host, status: 0, reported_at: DateTime.now.utc)
+
+    data['entries']&.map! do |entry|
```

Comment:
When `data['entries']` is `nil`, `PreupgradeReportEntry.create!(data['entries'])` on line 16 will raise an exception

```
ActiveRecord::RecordInvalid: Validation failed: Preupgrade report can't be blank, Hostname can't be blank, Title can't be blank, Actor can't be blank, Audience can't be blank, Severity can't be blank, Leapp run can't be blank
```

If you expect it might happen, wouldn't it be better to default to an empty array and use a variant of `map` which doesn't mutate element in place?